@here/harp-examples/src/camera-animations_key-track.ts

```diff
@@ -0,0 +1,220 @@
+ * generate animations.
+ *
+ * First we create a map.
+ * Then we generate different camera animations.
+ *
+ *
+ * Flying a bow to a next location using a maximum altitude for the bow. As a starting
+ * [[ControlPoint]], the current [[CameraPose]] is used and converted to a [[ControlPOint]].
+ * ```typescript
+ * [[include:harp_gl_camera_animation_1.ts]]
+ * ```
+ * An orbit animation around the current target of the map.
+ * ```typescript
+ * [[include:harp_gl_camera_animation_2.ts]]
+ * ```
+ *
+ * A [[CameraKeyTrackAnimation]] using some geoLocations to generate the [[ControlPoints]]
+ *
+ * The simple [[CameraKeyTrackAnimationOptions]] contain a list of [[ControlPoint]]s
+ * ```typescript
+ * [[include:harp_gl_camera_animation_3.ts]]
+ * ```
+ *
+ *
+ * A GUI element is added in the example to start the different animations
+ * ```typescript
+ * [[include:harp_gl_camera_animation_4.ts]]
+ * ```
+ */
+export namespace CameraAnimationExample {
+    // snippet:harp_gl_camera_animation_0.ts
+    const map = createBaseMap();
+    // end:harp_gl_camera_camera_animation_0.ts
+
+    const geoLocations: IData = {
+        Dubai: new GeoCoordinates(25.19705, 55.27419),
+        BerlinStation: new GeoCoordinates(52.5250871, 13.367208),
+        BerlinReichstag: new GeoCoordinates(52.5186234, 13.373993),
+        BerlinTower: new GeoCoordinates(52.52081829, 13.407225)
+    };
+
+    map.lookAt({
+        target: geoLocations.Dubai,
+        distance: 1000,
+        tilt: 45
+    });
+
+    const options = {
+        globe: true,
+        orbit: false,
+        flyTo: "",
+        flyOver: false
+    };
+    let cameraAnimation: CameraKeyTrackAnimation | undefined;
+
+    // snippet:harp_gl_camera_animation_3.ts
+    const flyOverAnimationOptions = {
+        controlPoints: [
+            new ControlPoint({
+                geoCoordinate: geoLocations.BerlinReichstag,
+                timestamp: 0,
+                azimuth: 300,
+                altitude: 45,
+                distance: 500
+            }),
+            new ControlPoint({
+                geoCoordinate: geoLocations.BerlinStation,
+                timestamp: 10,
+                azimuth: 20,
+                altitude: 45,
+                distance: 3000
+            }),
+            new ControlPoint({
+                geoCoordinate: geoLocations.BerlinTower,
+                timestamp: 15,
+                azimuth: 180,
+                altitude: 35,
+                distance: 200
+            })
+        ]
+    };
+    // end:harp_gl_camera_camera_animation_3.ts
+
+    const gui = new GUI({ width: 300 });
+    gui.add(options, "globe").onChange(() => {
+        map.projection = options.globe ? sphereProjection : mercatorProjection;
+    });
+    gui.add(options, "orbit")
+        .onChange(enableOrbit)
+        .listen();
+    gui.add(options, "flyTo", ["", "BerlinTower", "BerlinReichstag", "Dubai", "BerlinStation"])
```

Comment:
please generate the options from the keys of the `geoLocations` if not too difficult

Reply:
done